src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -957,5 +957,134 @@ public void Test_Issue1432()
                 });
             }
         }
+
+        [TestMethod]
+        public void Test_Issue1028_1()
```

Comment:
I think it would be worth breaking out these test methods into a dedicated new test class `KeyVaultSecretReferenceTests.cs` and making the method names a bit more descriptive - e.g. `KeyVault_secrets_can_be_passed_to_module_parameters()`, `KeyVault_secrets_cannot_be_exposed_as_module_outputs()` etc

Reply:
Yes, will do. I'll add a folder for scenario tests, as we'll be adding more and more separate files will reduce merge conflicts.

Reply:
moved to a file in folder called "Scenarios"